markdown/src/main/java/org/apache/zeppelin/markdown/PegdownWebSequencelPlugin.java

```diff
@@ -0,0 +1,147 @@
+ * Pegdown plugin for Websequence diagram.
+ */
+public class PegdownWebSequencelPlugin extends Parser implements BlockPluginParser {
+
+  public PegdownWebSequencelPlugin() {
+    super(PegdownParser.OPTIONS,
+        PegdownParser.PARSING_TIMEOUT_AS_MILLIS,
+        DefaultParseRunnerProvider);
+  }
+
+  public PegdownWebSequencelPlugin(Integer options,
+                                   Long maxParsingTimeInMillis,
+                                   ParseRunnerProvider parseRunnerProvider,
+                                   PegDownPlugins plugins) {
+    super(options, maxParsingTimeInMillis, parseRunnerProvider, plugins);
+  }
+
+  public static final String TAG = "%%%";
+
+  Rule startMarker() {
+    return Sequence(Spn1(), TAG, Sp(), "sequence", Sp());
+  }
+
+  String endMarker() {
+    return TAG;
+  }
+
+  Rule body() {
+    return OneOrMore(TestNot(TAG), BaseParser.ANY);
+  }
+
+  Rule block() {
+    StringBuilderVar style = new StringBuilderVar();
+    StringBuilderVar body = new StringBuilderVar();
+
+    return NodeSequence(
+        startMarker(),
+        Optional(
+            String("style="),
+            Sequence(OneOrMore(Letter()), style.append(match()), Spn1())),
+        Sequence(body(), body.append(match())),
+        endMarker(),
+        push(
+            new ExpImageNode("title",
+                createWebsequenceUrl(style.getString(), body.getString()),
+                new TextNode(""))));
+  }
+
+  public static String createWebsequenceUrl(String style,
+                                            String content) {
+
+    style = StringUtils.defaultString(style, "default");
+
+    OutputStreamWriter writer = null;
+    BufferedReader reader = null;
+
+    try {
+      String query = new StringBuilder()
+          .append("style=")
+          .append(style)
+          .append("&message=")
+          .append(URLEncoder.encode(content, "UTF-8"))
+          .append("&apiVersion=1")
+          .toString();
+
+      URL url = new URL("http://www.websequencediagrams.com");
+      URLConnection conn = url.openConnection();
+      conn.setDoOutput(true);
+      writer = new OutputStreamWriter(conn.getOutputStream(), StandardCharsets.UTF_8);
+      writer.write(query);
+      writer.flush();
+
+      StringBuilder response = new StringBuilder();
+      reader =
+          new BufferedReader(
+              new InputStreamReader(conn.getInputStream(), StandardCharsets.UTF_8));
+      String line;
+      while ((line = reader.readLine()) != null) {
+        response.append(line);
+      }
+
+      writer.close();
+      reader.close();
+
+      String json = response.toString();
+
+      int start = json.indexOf("?png=");
+      int end = json.indexOf("\"", start);
+
+      if (start != -1 && end != -1) {
+        return "http://www.websequencediagrams.com/" + json.substring(start, end);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to get proper response from websequencediagrams.com");
```

Comment:
It's always good idea to propagate e as cause though Constructor to aid debugging.